nexus/src/app/silo.rs

```diff
@@ -617,9 +620,11 @@ impl super::Nexus {
                 "passed password verification without a valid user"
             );
             let db_user = fetch_user.unwrap().1;
-            Ok(Some(db_user))
+            Ok(db_user)
         } else {
-            Ok(None)
+            Err(Error::Unauthenticated {
+                internal_message: "Failed password verification".to_string(),
+            })
```

Comment:
This was the other helper function. Instead of returning `Ok(None)` on auth failure and converting that to the error in the calling code, we return the appropriate error directly.